lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -3155,6 +3168,52 @@ static void rewriteIndirectApply(FullApplySite apply,
   }
 }
 
+// Rewrite checked_cast_branch instructions with loadable source and opaque target.
+static void rewriteCheckedCastBranch(CheckedCastBranchInst *checkedCastBranch,
+                                     AddressLoweringState &pass) {
+  auto loc = checkedCastBranch->getLoc();
+  auto *func = checkedCastBranch->getFunction();
+  auto *successBB = checkedCastBranch->getSuccessBB();
+  auto *failureBB = checkedCastBranch->getFailureBB();
+  auto srcVal = checkedCastBranch->getOperand();
+  auto termBuilder = pass.getTermBuilder(checkedCastBranch);
+  auto successBuilder = pass.getBuilder(successBB->begin());
+  auto failureBuilder = pass.getBuilder(failureBB->begin());
```

Comment:
Can you put all of these variables in a CheckedCastBrInfo struct and factor it with `visitCheckedCastBr`? Can the implementation also be factored?